release-notes/Cabal-3.14.0.0.md

```diff
@@ -0,0 +1,156 @@
+
+  ```haskell
+  findFileCwd
+    :: forall dir1 dir2 file
+     . Verbosity
+    -> Maybe (SymbolicPath CWD (Dir dir1))
+
+    -> [SymbolicPath dir1 (Dir dir2)]
+
+    -> RelativePath dir2 File
+
+    -> IO (SymbolicPath dir1 File)
+  ```
+
+  See Note [Symbolic paths] in `Distribution.Utils.Path` for further information
+  on the design of this API.
+
+- Add `MultilineStrings` extension [#10245](https://github.com/haskell/cabal/pull/10245)
+
+  - adds support for the `MultilineStrings` language extension (GHC proposal #637)
+
+- Add `NamedDefaults` extension [#9740](https://github.com/haskell/cabal/pull/9740)
+
+  - adds support for the `NamedDefaults` language extension (GHC proposal #409)
+
+
+### Other changes
+
+- Add flag ignore-build-tools [#10128](https://github.com/haskell/cabal/pull/10128)
+
+  - Adds flag --ignore-build-tools which allows a user to ignore the tool
+    dependencies declared in build-tool-depends. For general use, this flag
+    should never be needed, but it may be useful for packagers.
+
+- Do not try to build dynamic executables on Windows [#10217](https://github.com/haskell/cabal/pull/10217)
+
+  - Cabal will now exit with a descriptive error message instead of attempting to
+    build a dynamic executable on Windows.
+
+- Always pass `ghc-options` to GHC [#8717](https://github.com/haskell/cabal/pull/8717)
+
+  Previously, options set in the package field `ghc-options` would not be passed
+  to GHC during the link phase for shared objects (where multiple `.o` or
+  `.dyn_o` files are merged into a single object file). This made it impossible
+  to use `ghc-options` to use a different linker by setting (for example)
+  `ghc-options: -optl-fuse-ld=mold -optlm-fuse-ld=mold`; the options would be
+  dropped in the link phase, falling back to the default linker.
+
+  It was possible to work around this by duplicating the `ghc-options` to
+  `ghc-shared-options`, which _are_ passed in the shared link phase, but that had
+  the (undocumented and unfortunate) side-effect of disabling the GHC
+  `-dynamic-too` flag, effectively doubling compilation times when
+  `ghc-shared-options` are set.
+
+  Now, `ghc-options` are combined with `ghc-shared-options` (to accurately
+  reflect the documentation on this feature) and the fact that
+  `ghc-shared-options` disables `-dynamic-too` is documented.
+
+- Introduce SetupHooks [#9551](https://github.com/haskell/cabal/pull/9551)
+
+  Introduction of a new build type: Hooks.
+  This build type, intended as replacement to the Custom build type, integrates
+  better with the rest of the ecosystem (`cabal-install`, Haskell Language Server).
+
+  The motivation and full design of this new build-type are specified in the
+  Haskell Foundation Tech Proposal
+  [Replacing the Cabal Custom build-type](https://github.com/haskellfoundation/tech-proposals/pull/60).
+
+  Package authors willing to use this feature should declare `build-type: Hooks`
```

Comment:
Should this also mention that you need `Cabal-version: 3.14`?

Reply:
Yes.